### PR TITLE
OCPBUGSM-35151: validate HA properties on register cluster

### PR DESCRIPTION
**Pull request by danielerez**
# Assisted Pull Request

## Description

Validate UserManagedNetworking and VipDhcpAllocation when creating a cluster with None HighAvailabilityMode:
* User Managed Networking must be enabled on single node OpenShift:
  * If UserManagedNetworking not specified, should be set to true.
  * If false, return an API error.
* VIP DHCP Allocation cannot be enabled on single node OpenShift:
  * If VipDhcpAllocation not specified, should be set to false.
  * If true, return an API error.
 
## List all the issues related to this PR

- [ ] New Feature
- [x] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [ ] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [x] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @eranco74 
/cc @tsorya 
/cc @avishayt 
/cc @filanov 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -423,10 +423,27 @@ func (b *bareMetalInventory) validateRegisterClusterInternalParams(params *insta
 		if err != nil {
 			return common.NewApiError(http.StatusBadRequest, err)
 		}
-		log.Infof("HA mode is None, setting UserManagedNetworking to true and VipDhcpAllocation to false")
-		params.NewClusterParams.UserManagedNetworking = swag.Bool(true)
-		// in case of single node VipDhcpAllocation should be always false
-		params.NewClusterParams.VipDhcpAllocation = swag.Bool(false)
+
+		if params.NewClusterParams.UserManagedNetworking == nil {
```
Comment (tsorya) [inline internal/bminventory/inventory.go:427]: in case you add new checks please move it to the function
something like validateSingleNodeParams

Comment (tsorya) [reply internal/bminventory/inventory.go:427]: Please check that UI will not be broken

Comment (danielerez) [reply internal/bminventory/inventory.go:427]: sure, moved to a new func.

**Pull request merged**